src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SinkIslandCheck.java

```diff
@@ -222,6 +228,53 @@ private boolean edgeCharacteristicsToIgnore(final Edge edge)
                 // Ignore edges that have been way sectioned at the border, as has high probability
                 // of creating a false positive due to the sectioning of the way
                 || SyntheticBoundaryNodeTag.isBoundaryNode(edge.end())
-                || SyntheticBoundaryNodeTag.isBoundaryNode(edge.start());
+                || SyntheticBoundaryNodeTag.isBoundaryNode(edge.start())
+                // Ignore edges that are of type service and is connected to pedestrian navigable
+                // ways
+                || this.isServiceRoad(edge) && this.isConnectedToPedestrian(edge);
+    }
+
+    /**
+     * Checks if the edge is fully enclosed within areas that have amenity tags that are in the
+     * AMENITY_VALUES_TO_EXCLUDE list
+     *
+     * @param edge
+     *            any edge
+     * @return true if the edge is fully enclosed within the area with excluded amenity tag
+     */
+    private boolean isWithinAreasWithExcludedAmenityTags(final Edge edge)
+    {
+        return StreamSupport
```

Comment:
You should be able to fetch all areas that intersect your Edge first, then you can filter and check for a full enclosure